### PR TITLE
Fix upgrade migration of fields from members to themes

### PR DESCRIPTION
**Pull request by sbulen**
To address: Upgrade Issue - Can't get past Step 4 #3833

### CODE REVIEW

File: other/upgrade_2-1_mysql.sql [hunk 1]
```diff
@@ -1046,7 +1046,7 @@ if (!empty($select_columns))
 				'{db_prefix}themes',
 				array('id_member' => 'int', 'id_theme' => 'int', 'variable' => 'string', 'value' => 'string'),
 				$inserts,
-				array('id_theme', 'id_member', 'variable')
+				array('id_member', 'id_theme', 'variable', 'value')
```
Comment (albertlast) [inline other/upgrade_2-1_mysql.sql]: value field can't be a pk field.
and the pk is defined as 'id_theme', 'id_member', 'variable' -> i would keep the old arrengment

File: other/upgrade_2-1_mysql.sql [hunk 2]
```diff
@@ -1022,22 +1022,22 @@ if (!empty($select_columns))
 		while ($row = $smcFunc['db_fetch_assoc']($request))
 		{
 			if (!empty($row['aim']))
-				$inserts[] = array($row['id_member'], -1, 'cust_aolins', $row['aim']);
+				$inserts[] = array($row['id_member'], 0, 'cust_aolins', $row['aim']);
```
Comment (albertlast) [inline other/upgrade_2-1_mysql.sql]: -1 looks as the right default value (the datebase can handle -1)
the same for the other changes.

**Review by albertlast — CHANGES_REQUESTED** [review]

**Comment by albertlast** [comment]
Please consider that all mysql changes needs to be sync with pg upgrade when ther are right,
what in my eyes here not the true is.

File: other/upgrade_2-1_mysql.sql (hunk 2 above)
Comment (albertlast) [reply other/upgrade_2-1_mysql.sql]: i looked on the wrong col,
the -1 is in theme_id col...
here are -1 value not allowed -> 0 or even 1 could be right

File: other/upgrade_2-1_mysql.sql (hunk 1 above)
Comment (sbulen) [reply other/upgrade_2-1_mysql.sql]: OK.  Will revise & test.

**Comment by sbulen** [comment]
Added the pg updates.  I don't have pg, so I can't test them, but they're simple enough. 

I left the pk line alone.  That makes sense.  Thanks for the insight.  

FYI - I have tested with a theme_id of 0 as well as a theme_id of 1 here.  The theme_id of 0 I believe corresponds to the default, and is a little more consistent with what I read elsewhere in this code.  

When you use 0, the profiles of forum members are visible with this info throughout the forum, which I think is desired.

**Comment by albertlast** [comment]
Someone else had to say if theme_id 0 is a valid value for this context,
i only mention the database side.

File: other/upgrade_2-1_mysql.sql (hunk 2 above)
Comment (live627) [reply other/upgrade_2-1_mysql.sql]: https://github.com/SimpleMachines/SMF2.1/blob/release-2.1/Sources/Load.php#L1753

Probably should be `1`

Comment (sbulen) [reply other/upgrade_2-1_mysql.sql]: I think you're right...   Let me fix this...
I just added a bunch of these attributes manually online, & they all get added with a theme_id of 1.
I think the data migration should mimic the manual data entry, so I'll set it to 1.

**Comment by sbulen** [comment]
Changed to theme_id of 1, per discussion above.  Thanks for the input!

**Pull request merged**